dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -883,12 +883,26 @@ struct VmmListArgs {
 
 impl DbArgs {
     /// Run a `omdb db` subcommand.
+    ///
+    /// Mostly delegates to `run_cmd_inner`, taking care to properly terminate
+    /// the database connection.
     pub(crate) async fn run_cmd(
         &self,
         omdb: &Omdb,
         log: &slog::Logger,
     ) -> Result<(), anyhow::Error> {
         let datastore = self.db_url_opts.connect(omdb, log).await?;
+        let res = self.run_cmd_inner(&datastore, omdb, log).await;
```

Comment:
Take it or leave it: we might want to define `run_cmd_inner` as a function *within* the scope of `run_cmd`, as that will prevent it from being called outside `run_cmd`, even by other code within this module (which making it module-private doesn't achieve).  This might be worth doing if we want to be extra certain that nobody accidentally does the wrong thing.

See [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=be24d50f30e7a219b13e3cb89ccf4bff) demonstrating what I mean.

Reply:
Alternatively, `run_cmd_inner` _could_ just be an `async` block within `run_cmd` that we immediately `.await`, as early returns in the `async` block won't return from the outer function, just the async block ([example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1cd8f0674a9f4bea989ba96f6c573f6c)). That way, we don't need to have a whole other function for it. Up to you if you think that's nicer...